crates/sui-framework/packages/sui-system/sources/sui_system.move

```diff
@@ -239,9 +240,20 @@ module sui_system::sui_system {
         validator_address: address,
         ctx: &mut TxContext,
     ) {
+        let staked_sui = request_add_stake_non_entry(wrapper, stake, validator_address, ctx);
+        transfer::public_transfer(staked_sui, tx_context::sender(ctx));
+    }
+
+    /// The non-entry version of `request_add_stake`, which returns the staked SUI instead of transferring it to the sender.
+    public fun request_add_stake_non_entry(
```

Comment:
Not sure about naming...

Reply:
What do you recommend? We already have a `request_withdraw_stake_non_entry`....

Reply:
I also find the name a bit of a mouthful (though understood that picking a different naming convention will introduce an inconsistency with the existing `request_withdraw_stake_non_entry`)...

Going forward, we hopefully won't run into this because no one will use `entry` anymore :).

Reply:
I would suggest simply `public fun stake`